src/core/orderbookutils.js

```diff
@@ -71,7 +109,16 @@ OrderBookUtils.getOfferQuality = function(offer, currencyGets) {
       reference_date: new Date()
     });
   } else {
-    amount = Amount.from_json(offer.quality + IOU_SUFFIX);
+
+    let currency = offer.TakerGets.currency;
+    let issuer = offer.TakerGets.issuer;
+
+    if (currency === undefined || issuer === undefined) {
+      currency = offer.TakerPays.currency;
```

Comment:
True, these are just as arbitrary and pointless, no matter that they are at least taken from one of TakerPays / TakerGets.